### PR TITLE
Fix Baseplate Not Rendering & Generate Baseplate From Profile

### PR DESCRIPTION
**Pull request by EmperorArthur**
Incorporates #179

# Main Changes

First, this fixes the baseplate not correctly in [OpenSCAD](https://www.openscad.org/) version 2024.02.16 (git 55be9cf86).  Done by switching from `minovsky` to `offset` in one function.

Second, instead of using a negative of the bin's this creates the base plate using technical data from [the spec](https://gridfinity.xyz/specification/).

# Other changes

Deprecated `rounded_rectangle` in favor of a new `rounded_square` function.  Which operates like the built in `square` and `cube` functions.

## Opinion

The `minovsky` function just seems to cause more issues than it is worth.

### CODE REVIEW

**Comment by EmperorArthur** [comment]
Pictures from `python -m unittest`
only_counterbored_screw_holes_top
only_countersunk_screw_holes_bottom
only_countersunk_screw_holes_top
plain_magnet_holes_top
ribbed_magnet_holes
chamfered_and_ribbed_magnet_holes
chamfered_magnet_holes
magnet_and_counterbored_screw_holes_bottom
magnet_and_counterbored_screw_holes_top
magnet_and_countersunk_screw_holes_bottom
magnet_and_countersunk_screw_holes_top
magnet_holes_bottom
no_holes_bottom
no_holes_top
only_counterbored_screw_holes_bottom

**Comment by nirvdrum** [comment]
I tried out this PR for my first Gridfinity baseplate. It worked great. I quite like the crush ribs for keeping magnets in place. Thanks for pulling this together.

**Review by ndemarco — CHANGES_REQUESTED** [review]
Fit to drawer functionality has been lost.

The PR uses more modular construction (square_baseplate_lip, solid_square_baseplate, square_baseplate_corner), which appears to focus more on the modular and customizable aspect rather than ensuring the "fit to drawer" functionality.

**Comment by EmperorArthur** [comment]
> Fit to drawer functionality has been lost.
> 
> The PR uses more modular construction (square_baseplate_lip, solid_square_baseplate, square_baseplate_corner), which appears to focus more on the modular and customizable aspect rather than ensuring the "fit to drawer" functionality.

Fit to drawer functionality has been restored, though tests were not added.

Please re-examine.

**Review by ndemarco — COMMENTED** [review]
Works as intended. Excellent updates.

**Comment by EmperorArthur** [comment]
> Works as intended. Excellent updates.

Thanks!  It won't let me merge until you approve the re-review. :/

**Comment by freiser77** [comment]
@EmperorArthur : Thanks for your bugfix. I'm using your branch as a second remote origin and it solves #202  perfectly for me.

**Comment by EmperorArthur** [comment]
Tried manually merging this, but "main" is a protected branch, and requires one reviewer to approve.

**Comment by ndemarco** [comment]
Works as intended.

`Baseplates with `style` skeletonized and `style_hole` none is thicker than the same style with counterbore or countersink. The additional thickness when no screw holes are in place is unnecessary.

File: tests/openscad_runner.py
Comment (ndemarco) [inline tests/openscad_runner.py:1]: Nice.

**Review by ndemarco — APPROVED** [review]

**Review by Ruudjhuu — APPROVED** [review]

**Pull request merged**